src/Hazelcast.Net/Serialization/Compact/ReflectionSerializer.cs

```diff
@@ -528,5 +534,26 @@ public virtual void Write(ICompactWriter writer, object obj)
             for (var i = 0; i < x.Length; i++) r[i] = x.GetValue(i)?.ToString();
             return r;
         }
+
+        private static void VerifyTypeIsSupported(object o)
+        {
+            // for FullName to be null, the type would need to be derived from an open generic somehow,
+            // which makes no sense since it is the actual type of a concrete object. we can assume that
+            // type is not going to be null.
+            var type = o.GetType();
+            var name = type.FullName!;
```

Comment:
_Just trying to understand..._ Is there any possibility that the name could be  null?